test/hotspot/jtreg/vmTestbase/nsk/share/ClassUnloader.java

```diff
@@ -43,11 +44,14 @@
  *
  * <p>ClassUnloader mainly intends to unload a class which was loaded
  * with especial <code>ClassUnloader.loadClass()</code> method.
- * A class is considered unloaded if its class loader is finalized.
- * If there no finalization of class loader detected for some timeout,
- * class is considered still loaded and method returns <i>false</i>.
+ * A class is considered unloaded if its class loader is reclaimed.
+ * A runnable lambda is registered to JVM to be called whenever the
+ * class loader becomes unreachable.
+ * After setting the class loader to null, if within a timeout no notification
+ * of its reclaim is received, class is considered still loaded and
+ * <code>unloadClass()</code> method returns <i>false</i>.
```

Comment:
All of the existing commentary in this test is somewhat lacking in terms of grammatical correctness, but we can at least make improvements to the new text. I suggest the following:
```
 * A class is eligible for unloading if its class loader has been reclaimed.
 * A Cleaner is used to inform the main test code when the class loader
 * becomes unreachable and is reclaimed.
 * If, after setting the class loader to null, no notification that it has become
 * reclaimed is received within the timeout interval, then the class is considered
 * to still be loaded and <code>unloadClass()</code> returns <i>false</i>.
```